pkg/innerring/config.go

```diff
@@ -0,0 +1,260 @@
+	c.P2P.Ping.Interval, err = parseConfigDurationPositive(v, pingSection+"interval", "P2P ping interval")
+	if err != nil {
+		return c, o, err
+	}
+	c.P2P.Ping.Timeout, err = parseConfigDurationPositive(v, pingSection+"timeout", "P2P ping timeout")
+	if err != nil {
+		return c, o, err
+	}
+
+	o.Logger = _logger.Logger
+
+	return c, o, nil
+}
+
+var errMissingConfig = errors.New("config value is missing")
+
+func parseConfigUint64Condition(v *viper.Viper, key, desc string, cond func(uint64) error) (uint64, error) {
+	var res uint64
+	var err error
+	if !v.IsSet(key) {
+		err = errMissingConfig
+	}
+	if err == nil {
+		res, err = cast.ToUint64E(v.Get(key))
+		if err == nil && cond != nil {
+			err = cond(res)
+		}
+	}
+	if err != nil {
+		return res, fmt.Errorf("invalid %s '%s' (unsigned integer): %w", desc, key, err)
+	}
+	return res, nil
+}
+
+func parseConfigUint64Range(v *viper.Viper, key, desc string, min, max uint64) (uint64, error) {
+	return parseConfigUint64Condition(v, key, desc, func(val uint64) error {
+		if val < min || val > max {
+			return fmt.Errorf("out of allowable range [%d:%d]", min, max)
+		}
+		return nil
+	})
+}
+
+func parseConfigUint64Max(v *viper.Viper, key, desc string, max uint64) (uint64, error) {
+	return parseConfigUint64Range(v, key, desc, 0, max)
+}
+
+func parseConfigDurationCondition(v *viper.Viper, key, desc string, cond func(time.Duration) error) (time.Duration, error) {
+	var res time.Duration
+	var err error
+	if !v.IsSet(key) {
+		err = errMissingConfig
+	}
+	if err == nil {
+		res, err = cast.ToDurationE(v.Get(key))
+		if err == nil && cond != nil {
+			err = cond(res)
+		}
+	}
+	if err != nil {
+		return res, fmt.Errorf("invalid %s '%s' (duration): %w", desc, key, err)
+	}
+	return res, nil
+}
+
+func parseConfigDurationPositive(v *viper.Viper, key, desc string) (time.Duration, error) {
+	return parseConfigDurationCondition(v, key, desc, func(d time.Duration) error {
+		if d <= 0 {
+			return errors.New("must be positive")
+		}
+		return nil
+	})
+}
+
+func parseConfigStrings(v *viper.Viper, key, desc string) ([]string, error) {
+	var res []string
+	var err error
+	if !v.IsSet(key) {
+		err = errMissingConfig
+	}
+	if err == nil {
+		res, err = cast.ToStringSliceE(v.Get(key))
+	}
+	if err != nil {
+		return res, fmt.Errorf("invalid %s '%s' (string array): %w", desc, key, err)
+	}
+	return res, nil
+}
+
+func parseConfigPublicKeys(v *viper.Viper, key, desc string) (keys.PublicKeys, error) {
+	ss, err := parseConfigStrings(v, key, desc)
+	if err != nil {
+		return nil, err
+	}
+	res, err := keys.NewPublicKeysFromStrings(ss)
+	if err != nil {
+		return res, fmt.Errorf("invalid %s '%s' (public keys): %w", desc, key, err)
+	}
+	return res, nil
+}
+
+func parseConfigAddressesTCP(v *viper.Viper, key, desc string) ([]string, error) {
+	ss, err := parseConfigStrings(v, key, desc)
+	if err != nil {
+		return nil, err
+	}
+	for i := range ss {
+		_, err = net.ResolveTCPAddr("tcp", ss[i])
```

Comment:
NeoGo node is considered completely private and we do not inherit the entity completely. Otherwise, it would be better to completely inherit the node config to avoid duplication. So I don't consider it a regression, anyway many NeoGo settings are not configurable in the current variant.